course_info/tests/jasmine/controllers/DetailsControllerSpec.js

```diff
@@ -212,49 +214,337 @@ describe('Unit testing DetailsController', function () {
     });
         
     describe('fetchCourseInstanceDetails', function() {
+
+        var ci;
+        var members;
+        beforeEach(function () {
+            ci = {
+                course_instance_id: $routeParams.courseInstanceId,
+                title: 'Test Title',
+                short_title: 'Test',
+                sub_title: 'Jasime is your friend',
+                description: '<p>hello</p>',
+                sites: [
+                    {
+                        external_id: 'https://x.y.z/888',
+                        site_id: '888',
+                        map_type: 'official'
+                    },
+                    {
+                        external_id: 'https://x.y.z/999',
+                        site_id: '999',
+                        map_type: 'unofficial'
+                    }
+                ],
+                course: {
+                    school_id: 'abc',
+                    registrar_code: 'ILE-2222',
+                    registrar_code_display: '2222',
+                    course_id: '789',
+                    course_groups: [
+                        {
+                            name: 'group one',
+                            id: 1
+                        },
+                        {
+                            name: 'group two',
+                            id: 2
+                        }
+                    ],
+                    departments: [
+                        {
+                            name: 'dept1',
+                            id: 1
+                        }
+                    ]
+                },
+                term: {
+                    display_name: 'Summer 2015',
+                    academic_year: '2015',
+                },
+                primary_xlist_instances: [],
+            };
+
+            members = {
+                count: 100
+            };
+        });
+
         it('should handle both successful responses even if they arrive ' +
-            'separately');
+            'separately', function(){
+            courseInstances.instances[ci.course_instance_id] = ci;
+            dc = $controller('DetailsController', {$scope: scope});
+            spyOn(dc, 'handleCourseInstanceResponse');
```

Comment:
We don't use this spy, it can probably be removed. I think it's possibly too late to spy on it anyway -- if we were spying on this successfully and not using `and.callThrough` then line 281 (`expect(dc.courseInstance['title']).toEqual(ci.title)`) shouldn't work, because the real `handleCourseInstanceResponse()` wouldn't be called, no?